learn-data-structures-by-building-a-shopping-cart/index16.js

```diff
@@ -0,0 +1,60 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+}
+
+/*
+Create a method called applyDiscount. 
```

Comment:
This could be broken up into three steps. First to declare the method with the parameter, next to return the formula, and finally another to log a discounted price to the console.